Jellyfin.Api/Helpers/TranscodingJobHelper.cs

```diff
@@ -530,6 +531,15 @@ private void DeleteHlsPartialStreamFiles(string outputFilePath)
             {
                 var attachmentPath = Path.Combine(_appPaths.CachePath, "attachments", state.MediaSource.Id);
                 await _attachmentExtractor.ExtractAllAttachments(state.MediaPath, state.MediaSource, attachmentPath, CancellationToken.None).ConfigureAwait(false);
+
+                if (state.SubtitleStream.IsExternal)
+                {
+                    string subtitlePath = state.SubtitleStream.Path;
+                    string subtitlePathArgument = string.Format(CultureInfo.InvariantCulture, "file:\"{0}\"", subtitlePath.Replace("\"", "\\\"", StringComparison.Ordinal));
+                    string subtitleId = subtitlePath.GetMD5().ToString("N", CultureInfo.InvariantCulture);
+
+                    await _attachmentExtractor.ExtractAllAttachmentsExternal(subtitlePathArgument, subtitleId, attachmentPath, CancellationToken.None).ConfigureAwait(false);
```

Comment:
Why `CancellationToken.None`? Why not `cancellationTokenSource.Token`?

Reply:
Another thing. There is nothing to extract if it's a text based subtitle format. It is only specifically needed when the subtitle container is `.mks` (are there other containers that contain attachments?)

Reply:
Yeah, we could use `cancellationTokenSource.Token`, I didn't think about that. Although subtitle extraction should usually be almost instant and not cancellable. 
I changed it.

Reply:
I'm not sure if there are containers other than `.mks` where it's needed, probably not. I could check the file extension by getting it from `state.SubtitleStream.Path`, or is there a better way?

Reply:
I don't think there another container than mks. Matching the file extension should be fine.